coverage/expand.go

```diff
@@ -0,0 +1,363 @@
+				properties[k] = v
+			}
+		}
+		if referenceModel.Enum != nil {
+			output.Enum = referenceModel.Enum
+		}
+		if referenceModel.Type != nil {
+			output.Type = referenceModel.Type
+		}
+		if referenceModel.Format != nil {
+			output.Format = referenceModel.Format
+		}
+		if referenceModel.HasAdditionalProperties {
+			output.HasAdditionalProperties = true
+		}
+		if referenceModel.Bool != nil {
+			output.Bool = referenceModel.Bool
+		}
+		if referenceModel.IsReadOnly {
+			output.IsReadOnly = referenceModel.IsReadOnly
+		}
+		if referenceModel.IsRequired {
+			output.IsRequired = referenceModel.IsRequired
+		}
+		if referenceModel.Discriminator != nil {
+			output.Discriminator = referenceModel.Discriminator
+		}
+		if referenceModel.Variants != nil {
+			output.Variants = referenceModel.Variants
+		}
+		if referenceModel.Item != nil {
+			output.Item = referenceModel.Item
+		}
+	}
+
+	// expand properties
+	for k, v := range input.Properties {
+		//log.Println("[DEBUG] expand properties", k)
+		properties[k] = expandSchema(v, swaggerPath, fmt.Sprintf("%s.%s", modelName, k), identifier+"."+k, root, resolvedDiscriminator, resolvedModel)
+	}
+
+	// expand composition
+	for _, v := range input.AllOf {
+		//log.Println("[DEBUG] expand composition", v.Ref.String())
+		allOf := expandSchema(v, swaggerPath, fmt.Sprintf("%s.allOf", modelName), identifier, root, resolvedDiscriminator, resolvedModel)
+		if allOf.Properties != nil {
+			for k, v := range *allOf.Properties {
+				properties[k] = v
+			}
+		}
+	}
+
+	if len(properties) > 0 {
+		for _, v := range input.Required {
+			p, ok := properties[v]
+			if !ok {
+				log.Printf("[WARN] required property %s not found in %s", v, modelName)
+			}
+			p.IsRequired = true
+		}
+
+		// check if all properties are readonly
+		allReadOnly := true
+		for _, v := range properties {
+			if !v.IsReadOnly {
+				allReadOnly = false
+				break
+			}
+		}
+		if allReadOnly {
+			output.IsReadOnly = true
+		}
+
+		output.Properties = &properties
+	}
+
+	// expand items
+	if input.Items != nil {
+		//log.Println("[DEBUG] expand items", input.Items.Schema.Ref.String())
+		item := expandSchema(*input.Items.Schema, swaggerPath, fmt.Sprintf("%s[]", modelName), identifier+"[]", root, resolvedDiscriminator, resolvedModel)
+		output.Item = item
+	}
+
+	delete(resolvedModel, modelName)
+
+	// variants have circular reference
+	// expand variants
+	if input.Discriminator != "" {
+		_, hasResolvedDiscriminator := resolvedDiscriminator[modelName]
+		if !hasResolvedDiscriminator {
+			resolvedDiscriminator[modelName] = nil
+			//log.Println("[DEBUG] expand variants", modelName)
+			variants := make(map[string]*Model)
+
+			variantsTable, err := getVariantTable(swaggerPath)
+			if err != nil {
+				log.Fatalf("[ERROR] get variant table %s: %v", swaggerPath, err)
+			}
+			varSet, ok := variantsTable[modelName]
+			// level order traverse to find all variants
+			for ok && len(varSet) > 0 {
```

Comment:
>ok && [](http://example.com/codeflow?start=7&length=5)

Seems `ok` will not change. If so, please remove from `for`

Reply:
Thanks, I will change the code.